angular/projects/spark-angular/src/lib/components/sprk-tabs/sprk-tabs.component.spec.ts

```diff
@@ -0,0 +1,474 @@
+    testPanel3 = testElement.querySelectorAll('.sprk-c-Tabs__content')[2];
+
+    // Dynamic Test Component
+    dynamicTestFixture = TestBed.createComponent(DynamicComponent);
+    dynamicTestComponent = dynamicTestFixture.componentInstance;
+  });
+
+  it('should create itself', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should add classes if additionalClasses has a value', () => {
+    component.additionalClasses = 'sprk-u-man';
+    fixture.detectChanges();
+    expect(element.classList.toString()).toEqual('sprk-c-Tabs sprk-u-man');
+  });
+
+  it('should add tabstrip additional classes if tablistAdditionalClasses has a value', () => {
+    component.tablistAdditionalClasses = 'sprk-u-man';
+    fixture.detectChanges();
+    expect(element.querySelector('.sprk-c-Tabs__buttons').classList.toString()).toEqual('sprk-c-Tabs__buttons sprk-u-man');
+  });
+
+  it('should create an id for each tab', () => {
+    expect(testTab1.id).toContain('tabs');
+    expect(testTab2.id).toContain('tabs');
+    expect(testTab1.id).not.toEqual(testTab2.id);
+  });
+
+  it('should create an id for each panel', () => {
+    expect(testPanel1.id).toContain('tabs');
+    expect(testPanel2.id).toContain('tabs');
+    expect(testPanel1.id).not.toEqual(testPanel2.id);
+  });
+
+  it('should set aria-controls on each tab', () => {
+    expect(testTab1.getAttribute('aria-controls')).toEqual(testPanel1.id);
+    expect(testTab2.getAttribute('aria-controls')).toEqual(testPanel2.id);
+  });
+
+  it('should set aria-labelledby on each panel', () => {
+    expect(testPanel1.getAttribute('aria-labelledby')).toEqual(testTab1.id);
+    expect(testPanel2.getAttribute('aria-labelledby')).toEqual(testTab2.id);
+  });
+
+  it('should change which tab / panel is active if a non-active tab is clicked', () => {
+    testTab1.click();
+    expect(testTab1.getAttribute('aria-selected')).toEqual('true');
+    expect(testPanel1.classList.contains('sprk-u-HideWhenJs')).toEqual(false);
+  });
+
+  it('clicking the panel should do nothing to the active state', () => {
+    expect(testTab2.getAttribute('aria-selected')).toEqual('true');
+    expect(testPanel2.classList.contains('sprk-u-HideWhenJs')).toEqual(false);
+    testPanel1.click();
+    expect(testTab2.getAttribute('aria-selected')).toEqual('true');
+    expect(testPanel2.classList.contains('sprk-u-HideWhenJs')).toEqual(false);
+  });
+
+  it('resizing should set the aria-orientation correctly when window is >= 736', () => {
+    component.ariaOrientation(800, testElement);
+    expect(testElement.getAttribute('aria-orientation')).toEqual('horizontal');
+  });
+
+  it('resizing should set the aria-orientation correctly when <= 736', () => {
+    component.ariaOrientation(700, testElement);
+    expect(testElement.getAttribute('aria-orientation')).toEqual('vertical');
+  });
+
+  it('resizing should set the aria-orientation correctly', () => {
+    window.dispatchEvent(new Event('resize'));
```

Comment:
For this test to be reliable, before dispatching we would want to check that aria-orientation is not present as well.

Reply:
`aria-orientation` gets added in the constructor, so it's already on the element when this test runs. I rewrote this to check that resize events cause the `ariaOrientation` function to be called.